src/DIRACbenchmark/dirac_benchmark.py

```diff
@@ -0,0 +1,295 @@
+    ):
+        if it_1 == 1:
+            start = os.times()
+
+        # Now the iterations
+        for _j in xrange(iters):
+            t_1 = random.normalvariate(10, 1)
+            m_1 += t_1
+            m_2 += t_1 * t_1
+            p_1 += t_1
+            p_2 += t_1 * t_1
+
+        if it_1 == iterations_num:
+            end = os.times()
+            if measured_copies is not None:
+                # Reduce the total of running copies by one
+                measured_copies.value -= 1
+
+        it_1 += 1
+
+    cput = sum(end[:4]) - sum(start[:4])
+    wall = end[4] - start[4]
+
+    if not cput:
+        return None
+
+    # Return DIRAC-compatible values
+    return {
+        "CPU": cput,
+        "WALL": wall,
+        "NORM": calib * iterations_num / cput,
+        "UNIT": "DB12",
+    }
+
+
+def single_dirac_benchmark_process(result_object, iterations_num=1, measured_copies=None):
+
+    """Run singleDiracBenchmark() in a multiprocessing friendly way"""
+
+    benchmark_result = single_dirac_benchmark(
+        iterations_num=iterations_num, measured_copies=measured_copies
+    )
+
+    if not benchmark_result or "NORM" not in benchmark_result:
+        return
+
+    # This makes it easy to use with multiprocessing.Process
+    result_object.value = benchmark_result["NORM"]
+
+
+def multiple_dirac_benchmark(copies=1, iterations_num=1, extra_iter=False):
+
+    """Run multiple copies of the DIRAC Benchmark in parallel"""
+
+    processes = []
+    results = []
+
+    if extra_iter:
+        # If true, then we run one or more extra iterations in each
+        # copy until the number still being meausured is zero.
+        measured_copies = multiprocessing.Value("i", copies)
+    else:
+        measured_copies = None
+
+    # Set up all the subprocesses
+    for i in xrange(copies):
+        results.append(multiprocessing.Value("d", 0.0))
+        processes.append(
+            multiprocessing.Process(
+                target=single_dirac_benchmark_process,
+                args=(results[i], iterations_num, measured_copies),
+            )
+        )
+
+    # Start them all off at the same time
+    for process in processes:
+        process.start()
+
+    # Wait for them all to finish
+    for process in processes:
+        process.join()
+
+    raw = []
+    product = 1.0
+
+    for res in results:
+        raw.append(res.value)
+        product *= res.value
+
+    raw.sort()
+
+    # Return the list of raw results and various averages
+    return {
+        "raw": raw,
+        "copies": copies,
+        "sum": sum(raw),
+        "arithmetic_mean": sum(raw) / copies,
+        "geometric_mean": product ** (1.0 / copies),
+        "median": raw[(copies - 1) // 2],
+    }
+
+
+def wholenode_dirac_benchmark(copies=None, iterations_num=1, extra_iter=False):
+
+    """Run as many copies as needed to occupy the whole machine"""
+
+    # Try $MACHINEFEATURES first if not given by caller
```

Comment:
I would remove the `MACHINEFEATURES` references. The project unfortunately never really started.